src/modules/ods/ods.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable, InternalServerErrorException, NotFoundException } from '@nestjs/common';
+import { GetOdsCustomerResponse } from './dto/get-ods-customer-response.dto';
+import { InjectDataSource } from '@nestjs/typeorm';
+import { DATABASE } from '@ukef/constants';
+import { DataSource } from 'typeorm';
+import { PinoLogger } from 'nestjs-pino';
+import { ODS_ENTITIES, OdsEntity, odsStoredProcedureInput, odsStoredProcedureQueryParams } from './dto/ods-payloads.dto';
+
+@Injectable()
+export class OdsService {
+  constructor(
+    @InjectDataSource(DATABASE.ODS)
```

Comment:
This black box magic of adding the name of the data source to pick the right one is amazing, I don't fully understand it but I like it